### PR TITLE
feat: derive model/test owners from dbt group owner when no direct owner is set

### PR DESCRIPTION
**Pull request by arbiv**
## Summary
Projects that moved ownership from model-level `+owner` to dbt `+group` config lost ownership visibility in the Elementary schema: `dbt_models.owner` and, downstream, `dbt_tests.model_owners` → `elementary_test_results.owners` became `[]`. This breaks alert routing / node ownership, which read the `owners` field.

Root cause: owner resolution only ever read the node's direct owner:
```
raw_owner = meta.owner or config.owner   # never consulted the group's owner
```
Groups were captured (in `dbt_groups` + `group_name` columns) but their owner was never folded into `owner`.

This adds a **group-owner fallback**, following the email-first precedence already used by `dbt_groups` and keeping direct owner authoritative:
```
formatted_owner = split(meta.owner or config.owner)
if not formatted_owner and group_name:
    group_owner = get_group_owner(group_name)   # owner.email or owner.name from graph.groups
    if group_owner: formatted_owner.append(group_owner)
```

Because test `model_owners` derive from the tested model's flattened `owner`, this automatically flows into `elementary_test_results.owners` — no change needed in `flatten_test`.

### Changes
- New macro `elementary.get_group_owner(group_name)` — resolves a group's owner from `graph.groups`, preferring `owner.email`, falling back to `owner.name`.
- `flatten_model`: fall back to the group owner only when no direct model/config owner exists (direct owner still wins, mirroring dbt's config precedence).
- Integration tests in `test_groups.py`:
  - `test_model_owner_derived_from_group` — grouped model, no direct owner → `owner = [group email]`.
  - `test_test_owner_derived_from_group` — test on that model → `model_owners = [group email]`.
  - `test_direct_owner_takes_precedence_over_group` — direct owner wins, group ignored.

Ran locally against Postgres: the 3 new tests plus existing `test_groups.py` / `test_test_owners.py` (13 total) pass.

Link to Devin session: https://app.devin.ai/sessions/7cc0b65582e44ceb9b95e5ea9559f08d
Requested by: @arbiv

---

<a href="https://app.devin.ai/review/elementary-data/dbt-data-reliability/pull/1034" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * Added a new macro to look up group ownership from the dbt graph.
* **Bug Fixes**
  * Improved owner handling in dbt artifacts so models and tests can inherit ownership from their group when no direct owner is set.
  * Direct ownership now correctly takes precedence over group ownership when both are provided.
* **Tests**
  * Added coverage for owner parsing (including JSON/edge cases) and group-based ownership inheritance scenarios.

### CODE REVIEW

File: macros/utils/graph/get_group_owner.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{% macro get_group_owner(group_name) %}
+    {#
+        Resolve a dbt group's owner to a single string, preferring the owner email
+        and falling back to the owner name. Mirrors how dbt_groups exposes
+        owner_email / owner_name. Returns none when the group or owner is missing.
+    #}
+    {% if not group_name %} {% do return(none) %} {% endif %}
+    {% for group_node in graph.groups.values() %}
+        {% if group_node.get("resource_type") == "group" and group_node.get(
```
Comment (elazarlachkar) [inline macros/utils/graph/get_group_owner.sql]: Is `group_node.get("resource_type") == "group"` needed if we're iterating over `graph.groups`?

File: macros/utils/graph/get_group_owner.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            "name"
+        ) == group_name %}
+            {% set owner_dict = elementary.safe_get_with_default(
+                group_node, "owner", {}
+            ) %}
+            {% do return(owner_dict.get("email") or owner_dict.get("name")) %}
```
Comment (elazarlachkar) [inline macros/utils/graph/get_group_owner.sql:13]: Could it be a list, like the dict owner (we split it by `,`)?

**Review by elazarlachkar — COMMENTED** [review]
Wondering about the approach - as we sync both groups and models, perhaps we should do the fallback when using the data (in Elementary CLI)?
It will allow us to keep the "real" data as is, and we could change the fallbacks easily in the future.

**Review by elazarlachkar — APPROVED** [review]
